src/swap/utils/formatDecimals.ts

```diff
@@ -6,8 +8,55 @@ export function formatDecimals(
   inputInDecimals = true,
   decimals = 18,
 ): string {
+  // Input validation
+  if (typeof amount !== 'string' || amount.trim() === '') {
+    throw new Error('Invalid input: amount must be a non-empty string');
+  }
+  if (!Number.isInteger(decimals) || decimals < 0) {
+    throw new Error('Invalid input: decimals must be a non-negative integer');
+  }
+  if (!/^-?(?:0|[1-9]\d*)(?:\.\d+)?$/.test(amount)) {
+    throw new Error('Invalid input: amount must be a valid number string');
+  }
+  if (amount.startsWith('-')) {
+    throw new Error('Invalid input: negative numbers are not allowed');
+  }
```

Comment:
Do we need 2 checks for negative integers?

Reply:
updated the regex to check for negatives